Releases: add 0.16.1

harding opened the pull request:
Adds release notes and a short blog post.  Previews:

- Blog: http://dg0.dtrt.org/en/2018/06/13/release-0.16.1/
- Release notes: http://dg0.dtrt.org/en/releases/0.16.1/
- Download page: http://dg0.dtrt.org/en/download/

Todos (can be fixed up after merge):

- [x] Update files to correct dates for release:
    - `_releases/0.16.1` - change date in YAML header
    - `cd _posts/en/posts/ ; rename 's/.*release/'$(date -u -I)'-release/' *0.16.1.md ; git add .`
- [x] Add magnet URI for torrent
- [x] If release notes change, update the release file with the changes

Note: this PR is expected to fail Travis CI as it expects the binaries to be in /bin/bitcoin-core-0.16.1/, which they're not yet.  The CI should pass once if Travis is restarted after the binaries are in their final location.

jnewbery commented on _posts/en/posts/2018-06-07-release-0.16.1.md:
```diff
@@ -0,0 +1,51 @@
+
+- Removing a previously-deprecated option related to maximum block size
+  in favor of an option that sets maximum block weight.  This change
+  only affects miners.  The interaction between these two options
```
`The interaction between these two options sometimes confused miners, and this change simplifies the situation.` isn't isn't quite right. `-blockmaxsize` has been silently ignored since 0.15.1. This change simply makes it explicit that `-blockmaxsize` doesn't do anything.

jnewbery commented on _posts/en/posts/2018-06-07-release-0.16.1.md:
```diff
@@ -0,0 +1,51 @@
+  only affects miners.  The interaction between these two options
+  sometimes confused miners, and this change simplifies the situation.
+
+- Ceasing relay of transactions using a rarely-seen opcode for legacy
```
Perhaps explicitly name `OP_CODESEPARATOR`?

jnewbery commented on _posts/en/posts/2018-06-07-release-0.16.1.md:
```diff
@@ -0,0 +1,46 @@
+- Fixing a bug that could've potentially caused miners to lose revenue if
+  they produced two blocks in very rapid succession.
+
+- Ceasing relay of transactions using the rarely-seen `OP_CodeSeparator` opcode for legacy
```
Standard style for opcodes is all caps ie `OP_CODESEPERATOR`.

harding replied:
@jnewbery I'm aware that's the common practice, and I think it's silly.  Capitalizing everything makes both reading and writing harder for no good reason that I can see.

But if you'd like, I can change it.

Also, this absolutely should not be merged until the final binaries are in /bin, or all of the download links on the site will be broken.

MarcoFalke replied:
Agree with @jnewbery, best to just stick to the convention: https://www.google.com/search?q=OP_CODESEPERATOR

harding replied:
Changed to uppercase.  Commits and preview updated.  (But I continue to think we're painting the bikeshed an ugly color :-)  (Also, for some inexplicable reason I get two orders of magnitude more results on Google for the upper camel case version: https://www.google.com/search?q=OP_CodeSeparator )

MarcoFalke replied:
> Also, for some inexplicable reason I get two orders of magnitude more results on Google for the upper camel case version

The heuristic they use to determine the number of results is highly experimental ;)

MarcoFalke commented:
Needs rebase or merge conflict otherwise solved.

harding commented:
Updated release notes to the ones in the final v0.16.1 tag, squashed edit commits, and rebased to fix merge conflict.  Also updated the preview.  Confirmed this Travis build is only failing because of the missing binaries (which is the last test we run for that reason).

So when the binaries have been uploaded to /bin, the Travis tests can be re-run and this merged as soon as they pass.

jnewbery commented:
utACK 98cf0fce1fb0f091d4ce166b2da516b458fbec0f. Thanks David! The bikeshed looks great in its new coat of paint.

laanwj commented:
Thanks for already creating this!

laanwj commented:
magnet link is: `magnet:?xt=urn:btih:91069028aaf9f6bb3279e71bfd9ab164922e578e&dn=bitcoin-core-0.16.1&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Fzer0day.ch%3A1337&tr=udp%3A%2F%2Fexplodie.org%3A6969`
